portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -445,7 +445,19 @@ else if (pathArray.length == 3) {
 					0, fileName.indexOf(StringPool.QUESTION));
 			}
 
-			return DLAppServiceUtil.getFileEntry(groupId, folderId, fileName);
+			try {
+				return DLAppServiceUtil.getFileEntryByFileName(
```

Comment:
Should we invert the order of these operations?

I understand this new call is a special case when the fileName and fileEntry are different.

Reply:
With the new changes, the filename will be the prefered "name" to use to search the file.
And in some of the old cases, the title is the same that the fileName so with the new method we will get the FileEntry, so  less calls wil be, that is why I changed the order.